.github/workflows/build.yml

```diff
@@ -26,6 +26,9 @@ jobs:
             pip install -r test-requirements.txt
             pytest ./tests
 
+      - name: mypy
+        run: mypy .
```

Comment:
this can be an overkill, you gonna have lots and lot of things to fix to make it work.

I would recommend starting with something like ruff, slowly.

Reply:
Can we see what we actually get running it manually?